pkg/deploy/registry/deployconfig/strategy.go

```diff
@@ -63,11 +103,31 @@ func (strategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) field.
 	return validation.ValidateDeploymentConfigUpdate(obj.(*api.DeploymentConfig), old.(*api.DeploymentConfig))
 }
 
-// CheckGracefulDelete allows a build to be gracefully deleted.
+// CheckGracefulDelete allows a deployment config to be gracefully deleted.
 func (strategy) CheckGracefulDelete(obj runtime.Object, options *kapi.DeleteOptions) bool {
 	return false
 }
 
+// statusStrategy implements behavior for DeploymentConfig status updates.
+type statusStrategy struct {
+	strategy
+}
+
+var StatusStrategy = statusStrategy{Strategy}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update of status.
+func (statusStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newDc := obj.(*api.DeploymentConfig)
+	oldDc := old.(*api.DeploymentConfig)
+	newDc.Spec = oldDc.Spec
+	newDc.Labels = oldDc.Labels
```

Comment:
We agreed upstream that we should allow annotation changes in status updates. Let me link the discussion.